src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/CertificatePal.cs

```diff
@@ -40,20 +43,37 @@ public static ICertificatePal FromHandle(IntPtr handle)
         }
 
         /// <summary>
-        /// Returns the SafeCertContextHandle. Use this instead of FromHandle property when
+        /// Returns the SafeCertContextHandle. Use this instead of FromHandle() when
         /// creating another X509Certificate object based on this one to ensure the underlying
         /// cert context is not released at the wrong time.
         /// </summary>
         /// <param name="cert"></param>
         /// <returns></returns>
         public static ICertificatePal FromOtherCert(X509Certificate cert)
         {
-            CertificatePal newCert = (CertificatePal)FromHandle(cert.Handle);
-            newCert._certContextCloned = true;
+            bool success = false;
 
-            ((CertificatePal)cert.Pal)._certContextCloned = true;
+            CertificatePal originalPal = ((CertificatePal)cert.Pal);
+            originalPal._certContext.DangerousAddRef(ref success);
+            Debug.Assert(success, "originalPal AddRef");
```

Comment:
Unreachable: If DangerousAddRef fails it throws.  (The bool is so that if you DAR in the middle of a try, the finally knows if it needs to call Release)